src/local-development/deploy.ts

```diff
@@ -182,20 +177,49 @@ async function bulkDeploy(anyProjectPath: vscode.Uri) {
 
 			// Display errors
 			if (errors.length > 0) {
+				// Log errors
+				log('error', `Deployment finished with ${errors.length} errors:`)
+				for (const err of errors) {
+					log('error', ' - ' + deploymentErrorToString(err, false));
+				}
+
+				// Display the modal error dialog
+				const MAX_DISPLAYED_ERRORS = 4;
 				showErrorDialog(`Failed ${errors.length} of ${codesToDeploy.length} codes deployments`, {
 					modal: true,
-					detail: errors
-						.map(
-							(err) =>
-								`** ${err.componentType} "${err.componentLocalId}" - ${
-									err.codeType ? ' - ' + err.codeType : ''
-								} **\n${err.errorMessage}`,
-						)
-						.join('\n\n'),
+					detail:
+						// Display first X errors
+						errors
+							.filter((_value, index) => index < MAX_DISPLAYED_ERRORS)
+							.map((deploymentError) => deploymentErrorToString(deploymentError, true))
+							.join('\n\n') +
+						// + info "and Y other errors"
+						(errors.length > MAX_DISPLAYED_ERRORS
+							? `\n\n ... and ${errors.length - MAX_DISPLAYED_ERRORS} other errors.`
+							: ''),
 				});
 			} else if (codesToDeploy.length >= 2) {
```

Comment:
What if codesToDeploy  = 1?

Reply:
After a deeper investigation, I have improved this weird situation with the additional commit just now. :)